lib/ec2imgutils/ec2removeimg.py

```diff
@@ -0,0 +1,166 @@
+            return utils.find_images_by_name(owned_images, self.image_name)
+        elif self.image_name_fragment:
+            return utils.find_images_by_name_fragment(
+                owned_images,
+                self.image_name_fragment)
+        elif self.image_name_match:
+            try:
+                return utils.find_images_by_name_regex_match(
+                    owned_images,
+                    self.image_name_match)
+            except:
+                msg = 'Unable to complie regular expression "%s"'
+                msg = msg % self.image_name_match
+                raise EC2RemoveImgException(msg)
+        else:
+            msg = 'No deprecation image condition set. Should not reach '
+            msg += 'this point.'
+            raise EC2RemoveImgException(msg)
+
+    # ---------------------------------------------------------------------
+    def _get_snapshot_id(self, image):
+        """Get the snapshot id associated with this image"""
+        device_map = image.get('BlockDeviceMappings')
+        if not device_map:
+            error_msg = 'Image "%s" has no device map' % image.get('Name')
+            raise EC2RemoveImgException(error_msg)
+        # The image snapshot should be the first device
+        ebs_info = device_map[0].get('Ebs')
+        if not ebs_info:
+            error_msg = 'Image "%s" is not EBS backed'  % image.get('Name')
+            raise EC2RemoveImgException(error_msg)
+        snapshot_id = ebs_info.get('SnapshotId')
+        if not snapshot_id:
+            error_msg = 'No snapshot found for image "%s"'  % image.get('Name')
+            raise EC2RemoveImgException(error_msg)
+        return snapshot_id
+
+    # ---------------------------------------------------------------------
+    def print_remove_info(self):
+        """Print information about images that would be deleted"""
+        self._connect()
+        images = self._get_images_to_remove()
+        images_ok = self._check_images_boundary_condition(images)
+
+        if not images_ok:
+            # While the boundary conditions are not met, we are just printing
+            # information thus this is not considered a failure
+            return True
+        
+        header_msg = 'Would remove image '
+        if not self.keep_snap:
+            header_msg += 'and snapshot '
+        header_msg += 'in region: '
+        print(header_msg, self.region)
+        for image in images:
+            if not self.keep_snap:
+                snapshot = self._get_snapshot_id(image)
+                print('\t\t%s\t%s\t%s' % (
+                    image['ImageId'],
+                    image['Name'],
+                    snapshot
+                ))
+            else:
+                print('\t\t%s\t%s' % (image['ImageId'], image['Name']))
+
+        return True
+
+    # ---------------------------------------------------------------------
+    def remove_images(self):
+        """Remove the images"""
+        self._connect()
```

Comment:
Connect called but client not stored or used?

Reply:
Fixed now stored and used below. Thanks